lib/LedgerSMB/Sysconfig.pm

```diff
@@ -128,7 +128,11 @@ sub def {
         }
 
         ${$name} = $ENV{$envvar} if ( $envvar && defined $ENV{$envvar} );  # If an environment variable is associated and currently defined, override the configfile and default with the ENV VAR
-        $ENV{$envvar} = ${$name}                                # If an environment variable is associated Set it based on the current value (taken from the config file, default, or pre-existing env var.
+
+        # If an environment variable is associated, set it  based on the
+        # current value (taken from the config file, default, or pre-existing
+        #  env var.
+        local $ENV{$envvar} = ${$name}
```

Comment:
Basically, only this one and the one below are meant to modify global state, meaning that the others can be merged, but these two can't.